src/main/java/CurrentChargingRangeCount.java

```diff
@@ -0,0 +1,39 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class CurrentChargingRangeCount {
+	
+	int []  input = {3,3,5,4,10,11,12};
```

Comment:
data is hardcoded to be part of the class ReadingRangeCount.
shouldn't data be an external input? Also testing with various input ranges is not possible in this case!
Please correct.